### PR TITLE
Tools: add script returning build options

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: Tools/scripts/build_options.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    Feature('Other', 'TORQEEDO', 'HAL_TORQEEDO_ENABLED', 'Enable Torqeedo Motors', 0, None),
+    Feature('Other', 'RPM', 'RPM_ENABLED', 'Enable RPM sensors', 0, None),
+    Feature('Other', 'DISPLAY', 'HAL_DISPLAY_ENABLED', 'Enable I2C Displays', 0, None),
+    Feature('Other', 'GRIPPER', 'GRIPPER_ENABLED', 'Enable Gripper', 0, None),
+    Feature('Other', 'BEACON', 'BEACON_ENABLED', 'Enable Beacon', 0, None),
+    Feature('Other', 'LANDING_GEAR', 'LANDING_GEAR_ENABLED', 'Enable Landing Gear', 0, None),
+    Feature('Other', 'NMEA_OUTPUT', 'HAL_NMEA_OUTPUT_ENABLED', 'Enable NMEA Output', 0, None),
+    Feature('Other', 'BARO_WIND_COMP', 'HAL_BARO_WIND_COMP_ENABLED', 'Enable Baro Wind Compensation', 0, None),
+    Feature('Other', 'GENERATOR', 'HAL_GENERATOR_ENABLED', 'Enable Generator', 0, None),
+    Feature('Other', 'AC_OAPATHPLANNER', 'AC_OAPATHPLANNER_ENABLED', 'Enable Object Avoidance Path Planner', 0, None),
```
Comment (amilcarlucas) [inline Tools/scripts/build_options.py:99]:
```suggestion
    Feature('Safety', 'AC_OAPATHPLANNER', 'AC_OAPATHPLANNER_ENABLED', 'Enable Object Avoidance Path Planner', 0, None),
```

File: Tools/scripts/build_options.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    Feature('Other', 'SPRAYER', 'HAL_SPRAYER_ENABLED', 'Enable Sprayer', 0, None),
+    Feature('Other', 'TORQEEDO', 'HAL_TORQEEDO_ENABLED', 'Enable Torqeedo Motors', 0, None),
+    Feature('Other', 'RPM', 'RPM_ENABLED', 'Enable RPM sensors', 0, None),
+    Feature('Other', 'DISPLAY', 'HAL_DISPLAY_ENABLED', 'Enable I2C Displays', 0, None),
+    Feature('Other', 'GRIPPER', 'GRIPPER_ENABLED', 'Enable Gripper', 0, None),
+    Feature('Other', 'BEACON', 'BEACON_ENABLED', 'Enable Beacon', 0, None),
+    Feature('Other', 'LANDING_GEAR', 'LANDING_GEAR_ENABLED', 'Enable Landing Gear', 0, None),
+    Feature('Other', 'NMEA_OUTPUT', 'HAL_NMEA_OUTPUT_ENABLED', 'Enable NMEA Output', 0, None),
+    Feature('Other', 'BARO_WIND_COMP', 'HAL_BARO_WIND_COMP_ENABLED', 'Enable Baro Wind Compensation', 0, None),
+    Feature('Other', 'GENERATOR', 'HAL_GENERATOR_ENABLED', 'Enable Generator', 0, None),
```
Comment (amilcarlucas) [inline Tools/scripts/build_options.py:98]:
```suggestion
    Feature('ICE', 'GENERATOR', 'HAL_GENERATOR_ENABLED', 'Enable Generator', 0, None),
```

File: Tools/scripts/build_options.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    Feature('VTOL Frame', 'DECA', 'AP_MOTORS_FRAME_DECA_ENABLED', 'DECA', 0, None),
+    Feature('VTOL Frame', 'DODECAHEXA', 'AP_MOTORS_FRAME_DODECAHEXA_ENABLED', 'DODECAHEXA', 0, None),
+    Feature('VTOL Frame', 'Y6', 'AP_MOTORS_FRAME_Y6_ENABLED', 'Y6', 0, None),
+    Feature('VTOL Frame', 'OCTAQUAD', 'AP_MOTORS_FRAME_OCTAQUAD_ENABLED', 'OCTAQUAD', 0, None),
+
+    Feature('Other', 'SOARING', 'HAL_SOARING_ENABLED', 'Enable Soaring', 0, None),
+    Feature('Other', 'DEEPSTALL', 'HAL_LANDING_DEEPSTALL_ENABLED', 'Enable Deepstall Landing', 0, None),
+    Feature('Other', 'DSP',  'HAL_WITH_DSP', 'Enable DSP for In-Flight FFT', 0, None),
+    Feature('Other', 'SPRAYER', 'HAL_SPRAYER_ENABLED', 'Enable Sprayer', 0, None),
+    Feature('Other', 'TORQEEDO', 'HAL_TORQEEDO_ENABLED', 'Enable Torqeedo Motors', 0, None),
```
Comment (amilcarlucas) [inline Tools/scripts/build_options.py:90]:
```suggestion
    Feature('ESC', 'TORQEEDO', 'HAL_TORQEEDO_ENABLED', 'Enable Torqeedo Motors', 0, None),
    Feature('ESC', 'FETTEC_ONEWIRE', 'AP_FETTEC_ONEWIRE_ENABLED', 'Enable FETtec OneWire ESCs', 0, 'HAL_WITH_ESC_TELEM'),
    Feature('ESC', 'ESC_TELEMETRY', 'HAL_WITH_ESC_TELEM', 'Enable ESC telemetry', 0, None),
```

**Review by amilcarlucas — CHANGES_REQUESTED** [review]

File: Tools/scripts/build_options.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    Feature('VTOL Frame', 'QUAD', 'AP_MOTORS_FRAME_QUAD_ENABLED', 'QUADS(BI,TRI also)', 1, None),
+    Feature('VTOL Frame', 'HEXA', 'AP_MOTORS_FRAME_HEXA_ENABLED', 'HEXA', 0, None),
+    Feature('VTOL Frame', 'OCTA', 'AP_MOTORS_FRAME_OCTA_ENABLED', 'OCTA', 0, None),
+    Feature('VTOL Frame', 'DECA', 'AP_MOTORS_FRAME_DECA_ENABLED', 'DECA', 0, None),
+    Feature('VTOL Frame', 'DODECAHEXA', 'AP_MOTORS_FRAME_DODECAHEXA_ENABLED', 'DODECAHEXA', 0, None),
+    Feature('VTOL Frame', 'Y6', 'AP_MOTORS_FRAME_Y6_ENABLED', 'Y6', 0, None),
+    Feature('VTOL Frame', 'OCTAQUAD', 'AP_MOTORS_FRAME_OCTAQUAD_ENABLED', 'OCTAQUAD', 0, None),
+
+    Feature('Other', 'SOARING', 'HAL_SOARING_ENABLED', 'Enable Soaring', 0, None),
```
Comment (amilcarlucas) [inline Tools/scripts/build_options.py:86]:
```suggestion
    Feature('Plane', 'SOARING', 'HAL_SOARING_ENABLED', 'Enable Soaring', 0, None),
```

File: Tools/scripts/build_options.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    Feature('Other', 'DEEPSTALL', 'HAL_LANDING_DEEPSTALL_ENABLED', 'Enable Deepstall Landing', 0, None),
```
Comment (amilcarlucas) [inline Tools/scripts/build_options.py:87]:
```suggestion
    Feature('Plane', 'DEEPSTALL', 'HAL_LANDING_DEEPSTALL_ENABLED', 'Enable Deepstall Landing', 0, None),
```

File: Tools/scripts/build_options.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    Feature('Other', 'DSP',  'HAL_WITH_DSP', 'Enable DSP for In-Flight FFT', 0, None),
+    Feature('Other', 'SPRAYER', 'HAL_SPRAYER_ENABLED', 'Enable Sprayer', 0, None),
```
Comment (amilcarlucas) [inline Tools/scripts/build_options.py:89]:
```suggestion
    Feature('Payload', 'SPRAYER', 'HAL_SPRAYER_ENABLED', 'Enable Sprayer', 0, None),
```

File: Tools/scripts/build_options.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    Feature('Other', 'TORQEEDO', 'HAL_TORQEEDO_ENABLED', 'Enable Torqeedo Motors', 0, None),
+    Feature('Other', 'RPM', 'RPM_ENABLED', 'Enable RPM sensors', 0, None),
+    Feature('Other', 'DISPLAY', 'HAL_DISPLAY_ENABLED', 'Enable I2C Displays', 0, None),
+    Feature('Other', 'GRIPPER', 'GRIPPER_ENABLED', 'Enable Gripper', 0, None),
```
Comment (amilcarlucas) [inline Tools/scripts/build_options.py:93]:
```suggestion
    Feature('Payload', 'GRIPPER', 'GRIPPER_ENABLED', 'Enable Gripper', 0, None),
```

File: Tools/scripts/build_options.py [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    Feature('Other', 'BEACON', 'BEACON_ENABLED', 'Enable Beacon', 0, None),
+    Feature('Other', 'LANDING_GEAR', 'LANDING_GEAR_ENABLED', 'Enable Landing Gear', 0, None),
+    Feature('Other', 'NMEA_OUTPUT', 'HAL_NMEA_OUTPUT_ENABLED', 'Enable NMEA Output', 0, None),
+    Feature('Other', 'BARO_WIND_COMP', 'HAL_BARO_WIND_COMP_ENABLED', 'Enable Baro Wind Compensation', 0, None),
+    Feature('Other', 'GENERATOR', 'HAL_GENERATOR_ENABLED', 'Enable Generator', 0, None),
+    Feature('Other', 'AC_OAPATHPLANNER', 'AC_OAPATHPLANNER_ENABLED', 'Enable Object Avoidance Path Planner', 0, None),
+    Feature('Other', 'WINCH', 'WINCH_ENABLED', 'Enable Winch', 0, None),
```
Comment (amilcarlucas) [inline Tools/scripts/build_options.py:100]:
```suggestion
    Feature('Payload', 'WINCH', 'WINCH_ENABLED', 'Enable Winch', 0, None),
```

**Review by amilcarlucas — CHANGES_REQUESTED** [review]

**Comment by peterbarker** [comment]
@amilcarlucas thanks for the review.

I would prefer not to change the data I'm moving from the other tree as part of this PR.  I think your suggestions there are good - maybe another PR against master once this is merged?

**Comment by amilcarlucas** [comment]
OK.

File: Tools/scripts/build_options.py (hunk 1 above)
Comment (khancyr) [reply Tools/scripts/build_options.py:99]: Avoidance/avoid, but not safety !

**Comment by Hwurzburg** [comment]
Thanks for completing this

**Comment by peterbarker** [comment]
@Hwurzburg were you able to test?

**Review by tridge — APPROVED** [review]

**Pull request merged**